pkg/grpc/server_test.go

```diff
@@ -19,15 +21,42 @@ import (
 	"github.com/stackrox/rox/pkg/grpc/authz/allow"
 	"github.com/stackrox/rox/pkg/grpc/routes"
 	"github.com/stackrox/rox/pkg/mtls/verifier"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 	"google.golang.org/grpc"
 )
 
-const (
-	testDefaultPort = 8080
+// Starting with 50000, because Dynamic and/or Private Ports are (49152-65535).
+const minFreePortRange = 50000
+const maxFreePortRange = 60000
+
+var (
+	once        sync.Once
+	portCounter atomic.Uint64
 )
 
+func getFreeTestPort() uint64 {
+	once.Do(func() {
+		portCounter.Store(minFreePortRange)
+	})
+
+	for {
+		freePort := portCounter.Add(1)
+		if freePort > maxFreePortRange {
+			panic("port number is out of range")
```

Comment:
[nit] maybe here it would be better to fail the test instead of panicking.

Reply:
tbh. I would prefer to keep it as it is. I want to keep the interface (function signature) simple. If we ever face `panic` here (10k ports exhausted) - I would prefer to change implementation and start cycling ports from the beginning. That would introduce some complexity, and maybe there is no need for that.